code/mob.dm

```diff
@@ -3026,3 +3026,7 @@
 		..(0)
 	else
 		..(newdensity)
+
+// to check if someone is abusing cameras with stuff like artifacts, power gloves, etc
+/mob/proc/in_real_view_range(var/turf/T)
+	return src.client && get_dist(T,src) <= (istext(src.client.view) ? 10 : src.client.view)
```

Comment:
use `IN_RANGE` macro perhaps (handles z-levels etc)

Reply:
Thanks, good idea!

Also special shoutout to byond for requiring parentheses even though there is no logical reason why you would need parentheses there and what the fuck why are you returning the string to me